pkg/core/object/fmt.go

```diff
@@ -158,17 +158,17 @@ func (v *FormatValidator) Validate(obj *object.Object, unprepared bool) error {
 	}
 
 	if !unprepared {
+		if err := obj.CheckHeaderVerificationFields(); err != nil {
+			return fmt.Errorf("could not validate header fields: %w", err)
+		}
+
 		if err := validateSignatureKey(obj); err != nil {
 			return fmt.Errorf("could not validate signature key: %w", err)
 		}
 
 		if err := v.checkExpiration(*obj); err != nil {
 			return fmt.Errorf("object did not pass expiration check: %w", err)
 		}
-
-		if err := obj.CheckHeaderVerificationFields(); err != nil {
```

Comment:
I agree that it has precedence over the expiration check, however its relation to `validateSignatureKey` is far less obvious. The chain of trust is "container owner -> session key -> object signature" and I'd expect following this chain for any checks.

At the same time _format_ checks have precedence over signature checks, currently they're somewhat mixed (related to following commits).

Reply:
kept the order